deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java

```diff
@@ -104,6 +114,20 @@
 
   @Override
   public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    try {
+      processInternal(annotations, roundEnv);
+    } catch (DeepLinkProcessorException e) {
+      error(e.getElement(), e.getMessage());
+    }
+    return true;
+  }
+
+  private void processInternal(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    if (getCustomAnnotations().length == 0) {
```

Comment:
it should be fine to have this be empty if no custom annotations have been defined. 

Can we move this check to line 134, and when adding a custom annotation element make sure the type is already defined in `getCustomAnnotations()`?

Reply:
By the way, just to clarify with how the incremental processor works (took me ages to get my head around it). You need to be specific about what annotations your processor handles. if you do not specify an annotation, the processor will never be told about it. 

Therefore the `annotation.getAnnotation(DEEP_LINK_SPEC_CLASS)` code would always return a null value.